### PR TITLE
Ch8569/municipal team construction

### PR DESCRIPTION
**Pull request by ktherage**

### CODE REVIEW

File: src/Controller/EnMarche/MunicipalChiefController.php [hunk 1]
```diff
@@ -106,17 +105,129 @@ public function municipalRunningMateDetailAction(
...
+        $this->disableInProduction();
+
+        $this->addToTeam($objectManager, $volunteerRequest, $municipalChief->getMunicipalChiefManagedArea());
+
+        return $this->redirectToRoute('app_municipal_chief_municipal_volunteer_request');
+    }
+
+    /**
+     * @Route(
+     *     path="/municipale/candidature-benevole/{uuid}/retirer-a-mon-equipe",
```
Comment (ottaviano) [inline src/Controller/EnMarche/MunicipalChiefController.php]: `retirer-de-mon...`

File: src/Controller/EnMarche/MunicipalChiefController.php [hunk 2]
```diff
@@ -106,17 +105,129 @@ public function municipalRunningMateDetailAction(
...
+        $this->disableInProduction();
+
+        $this->addToTeam($objectManager, $runningMateRequest, $municipalChief->getMunicipalChiefManagedArea());
+
+        return $this->redirectToRoute('app_municipal_chief_municipal_running_mate_request');
+    }
+
+    /**
+     * @Route(
+     *     path="/municipale/candidature-colistiers/{uuid}/retirer-a-mon-equipe",
```
Comment (ottaviano) [inline src/Controller/EnMarche/MunicipalChiefController.php]: `retirer-de-mon...`

File: src/Controller/EnMarche/MunicipalChiefController.php (hunk 1 above)
Comment (Remg) [reply src/Controller/EnMarche/MunicipalChiefController.php]: :smirk:

File: src/Controller/EnMarche/MunicipalChiefController.php [hunk 3]
```diff
@@ -106,17 +105,129 @@ public function municipalRunningMateDetailAction(
...
+            $municipalChiefManagedArea->getCodes(),
+            $applicationRequest->getFavoriteCities()
+        );
+        if (!$applicationRequest->getTakenForCity() && !empty($city = reset($cities))) {
+            $applicationRequest->setTakenForCity($city);
+            $objectManager->flush();
+        }
+    }
+
+    private function removeToTeam(
```
Comment (Remg) [inline src/Controller/EnMarche/MunicipalChiefController.php]: `removeFromTeam`

File: src/Entity/ApplicationRequest/ApplicationRequest.php [hunk 4]
```diff
@@ -413,4 +420,14 @@ public function getFavoriteThemesAsString(): string
 
         return implode(', ', $themes);
     }
+
+    public function getTakenForCity(): ?string
+    {
+        return $this->takenForCity;
+    }
+
+    public function setTakenForCity(?string $takenForCity = null): void
```
Comment (Remg) [inline src/Entity/ApplicationRequest/ApplicationRequest.php]: a setter should never be called with `no` argument.

so `(?string $takenForCity)` would be enough. (no need a default value if no parameter is given)

File: front/style/manager-space/roles/_referent.scss [hunk 5]
```diff
@@ -102,6 +102,12 @@
 
 }
 
+span.disabled {
```
[the remark this answers is not in the document]
Comment (ktherage) [reply front/style/manager-space/roles/_referent.scss]: @ottaviano hum, it's a span, wrapping a link itself vrappinng an icon. i couldn't use it.

File: src/Controller/EnMarche/MunicipalChiefController.php (hunk 1 above)
Comment (ktherage) [reply src/Controller/EnMarche/MunicipalChiefController.php]: :sweat_smile:

File: front/style/manager-space/roles/_referent.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (ottaviano) [reply front/style/manager-space/roles/_referent.scss]: why not? did you try to replace: 
```html
<span id='application-team-$i' class='btn btn--default".(!$canRemove ? ' disabled' : '')."'>
  ...
</span>
```
by :
```html
<span id='application-team-$i' class='btn btn--default".(!$canRemove ? ' btn-disabled' : '')."'>
  ...
</span>
```

**Pull request merged**